File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.py
def solution(n):
    answer = 0
    oddanswer = 0
    evenanswer = 0
    if (n%2!=0) :
        for i in range(n+1):
            if(i%2!=0) :
                oddanswer += i
        
        answer = oddanswer
    else :
        for i in range(n+1):
            if(i%2==0) :
                evenanswer+= i*i
        
        answer = evenanswer
    return answer

File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.js
function solution(n) {
    var answer = 0;
    let evensum = 0;
    let oddsum=0;
    for(let i=1; i<=n ; i++){
        if(i%2==0) {
            evensum+=(i*i);
        }
        else {
            oddsum+=i;
        }
    }
    answer = n%2==0 ? evensum : oddsum;
    return answer;
}